### PR TITLE
Linux workflow stat-o-dynamic

### PR DESCRIPTION
**Pull request by nirvn**
This PR moves the last remaining workflow that relied system Qt libraries - namely linux - to self-built ones, which means we can get rid of the whole vcpkg/overlay_system_qt6 folder, and rely solely on vcpkg revision changes to update Qt across all our platforms.

Since completely static linux built would lead to linuxdeploy failure, and because we don't want to spend _too much time_ debugging root causes here, I've moved the linux build to match that of Android, i.e. mostly dynamic with a list of ports built statically. One bonus here: our linux becomes a much closer mirror to what we ship on Android, which can help debug things.

On the print support front, when building Qt on linux, linuxdeploy is looking for the cups print support plugin. It is shipped by Qt's pre-built libraries, so we're merely copying that behavior here.

@m-kuhn , the ball is in your court :)

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6353fdf0a0204374f724af1f30988bda2e3847ee: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4917-1a36fc2d93959c817a78317d14a1db84c75d771e-x86-android.apk)
</details>

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -73,7 +64,7 @@ jobs:
                 -D WITH_VCPKG=ON \
                 -D WITH_SPIX=ON \
                 -D WITH_NFC=OFF \
-                -D SYSTEM_QT=ON \
+                -D SYSTEM_QT=OFF \
```
Comment (m-kuhn) [inline .github/workflows/linux.yml:67]: Can we eliminate this altogether?

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -87,7 +87,7 @@ jobs:
       - name: 🔨 Prepare build env
         run: |
           sudo apt-get update
-          sudo apt-get install -y gperf autopoint '^libxcb.*-dev' libx11-xcb-dev libegl1-mesa libegl1-mesa-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev autoconf-archive libgstreamer-gl1.0-0 libgstreamer-plugins-base1.0-0 libfuse2 nasm
+          sudo apt-get install -y gperf autopoint '^libxcb.*-dev' libx11-xcb-dev libegl1-mesa libegl1-mesa-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev autoconf-archive libgstreamer-gl1.0-0 libgstreamer-plugins-base1.0-0 libfuse2 libcups2-dev nasm
```
Comment (m-kuhn) [inline .github/workflows/android.yml:90]: Could this be built as part of vcpkg or not?

**Review by m-kuhn — APPROVED** [review]
Nice work

File: .github/workflows/linux.yml (hunk 1 above)
Comment (nirvn) [reply .github/workflows/linux.yml:67]: Yes

File: .github/workflows/android.yml (hunk 2 above)
Comment (nirvn) [reply .github/workflows/android.yml:90]: Maybe, but it's not there ATM.

An alternative would be to add a proper WITH_PRINTER=ON/OFF in QGIS, which would remove the need for printing altogether which we do not use.

If we want to spend more energy, I'd vote for the latter.

Comment (m-kuhn) [reply .github/workflows/android.yml:90]: Let's stick to what we have here

File: .github/workflows/linux.yml (hunk 1 above)
Comment (nirvn) [inline .github/workflows/linux.yml:67]:
```suggestion
```

Comment (nirvn) [reply .github/workflows/linux.yml]: Done

**Pull request merged**